src/Moq/EventHandlerCollection.cs

```diff
@@ -3,33 +3,23 @@
 
 using System;
 using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
 
 namespace Moq
 {
 	internal sealed class EventHandlerCollection
 	{
-		private Dictionary<string, List<Delegate>> eventHandlers;
+		private Dictionary<string, Delegate> eventHandlers;
 
 		public EventHandlerCollection()
 		{
-			this.eventHandlers = new Dictionary<string, List<Delegate>>();
+			this.eventHandlers = new Dictionary<string, Delegate>();
 		}
 
 		public void Add(string eventName, Delegate eventHandler)
 		{
 			lock (this.eventHandlers)
 			{
-				List<Delegate> handlers;
-				if (!this.eventHandlers.TryGetValue(eventName, out handlers))
-				{
-					handlers = new List<Delegate>();
-					this.eventHandlers.Add(eventName, handlers);
-				}
-
-				handlers.Add(eventHandler);
+				this.eventHandlers[eventName] = Delegate.Combine(this.TryGet(eventName), eventHandler);
```

Comment:
Calling the public `TryGet` method will take the same lock once more. Try to prevent this. Same in `Remove`.